File: app/src/main/java/de/nicidienase/chaosflix/touch/fragments/ConferenceRecyclerViewAdapter.java
package de.nicidienase.chaosflix.touch.fragments;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import org.w3c.dom.Text;

import de.nicidienase.chaosflix.R;
import de.nicidienase.chaosflix.shared.entities.recording.Conference;
import de.nicidienase.chaosflix.touch.fragments.ConferenceFragment.OnListFragmentInteractionListener;

import java.util.List;

public class ConferenceRecyclerViewAdapter extends RecyclerView.Adapter<ConferenceRecyclerViewAdapter.ViewHolder> {

	private final List<Conference> mConferences;
	private final OnListFragmentInteractionListener mListener;

	public ConferenceRecyclerViewAdapter(List<Conference> items, OnListFragmentInteractionListener listener) {
		mConferences = items;
		mListener = listener;
	}

	@Override
	public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		View view = LayoutInflater.from(parent.getContext())
				.inflate(R.layout.fragment_conference, parent, false);
		return new ViewHolder(view);
	}

	@Override
	public void onBindViewHolder(final ViewHolder holder, int position) {
		holder.mItem = mConferences.get(position);
		holder.mTitleText.setText(mConferences.get(position).getTitle());
		holder.mAcronym.setText(mConferences.get(position).getAcronym());
		Glide.with(holder.mIcon.getContext())
				.load(mConferences.get(position).getLogoUrl())
				.fitCenter()
				.into(holder.mIcon);

		holder.mView.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				if (null != mListener) {
					// Notify the active callbacks interface (the activity, if the
					// fragment is attached to one) that an item has been selected.
					mListener.onListFragmentInteraction(holder.mItem);
				}
			}
		});
	}

	@Override
	public int getItemCount() {
		return mConferences.size();
	}

	public class ViewHolder extends RecyclerView.ViewHolder {
		public final View mView;
		public final ImageView mIcon;
		public final TextView mTitleText;
		public final TextView mAcronym;
		public Conference mItem;

		public ViewHolder(View view) {
			super(view);
			mView = view;
			mIcon = (ImageView) view.findViewById(R.id.imageView);
			mTitleText = (TextView) view.findViewById(R.id.title_text);
			mAcronym = (TextView) view.findViewById(R.id.acronym_text);
		}

		@Override
		public String toString() {
			return super.toString() + " '" + mTitleText.getText() + "'";
		}
	}
}

File: touch/src/main/java/de/nicidienase/chaosflix/touch/browse/adapters/ConferenceRecyclerViewAdapter.kt
package de.nicidienase.chaosflix.touch.browse.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil.ItemCallback
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import de.nicidienase.chaosflix.common.mediadata.entities.recording.persistence.Conference
import de.nicidienase.chaosflix.touch.databinding.ItemConferenceCardviewBinding

class ConferenceRecyclerViewAdapter(private val mListener: (Conference) -> Unit) : ListAdapter<Conference, ConferenceRecyclerViewAdapter.ViewHolder>(conferenceDiffUtil) {

    override fun getItemId(position: Int): Long {
        return getItem(position).id
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemConferenceCardviewBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.binding.conference = getItem(position)
        holder.binding.root.setOnClickListener { _ ->
            mListener(getItem(position))
        }
    }

    class ViewHolder(val binding: ItemConferenceCardviewBinding) : RecyclerView.ViewHolder(binding.root)

    companion object {
        private val conferenceDiffUtil = object : ItemCallback<Conference>() {
            override fun areItemsTheSame(oldItem: Conference, newItem: Conference): Boolean = oldItem === newItem
            override fun areContentsTheSame(oldItem: Conference, newItem: Conference): Boolean = oldItem.url == newItem.url
        }
    }
}
